frontend/src/Components/ThemeToggleComponent.tsx

```diff
@@ -0,0 +1,53 @@
+import buttonStyles from '../Components/Buttons.module.scss';
+import React, {useEffect} from 'react';
+import {ReactComponent as DarkIcon} from '../Assets/theme_dark.svg';
+import {ReactComponent as LightIcon} from '../Assets/theme_light.svg';
+import {useTheme} from '../Theme/ThemeProvider';
+import classNames from 'classnames';
+
+interface ThemeToggleComponentProps {
+  buttonLabel?: string;
+  resetOnOnmount?: boolean;
+  dynamic?: boolean;
+}
+
+export default function ThemeToggleComponent(props: ThemeToggleComponentProps) {
+  const {theme, setTheme} = useTheme();
+  let initialTheme: string;
+
+  useEffect(() => {
+    if (theme) {
+      initialTheme = theme;
+      return () => {
```

Comment:
☝️ what she said.

If the intention is to remember the original value of the `theme`, just create another state variable:

```
const [initialTheme] = useState(theme);
```